overlord/ifacestate/handlers.go

```diff
@@ -742,7 +745,53 @@ func (m *InterfaceManager) undoConnect(task *state.Task, _ *tomb.Tomb) error {
 	}
 	setConns(st, conns)
 
-	return m.repo.Disconnect(connRef.PlugRef.Snap, connRef.PlugRef.Name, connRef.SlotRef.Snap, connRef.SlotRef.Name)
+	if err := m.repo.Disconnect(connRef.PlugRef.Snap, connRef.PlugRef.Name, connRef.SlotRef.Snap, connRef.SlotRef.Name); err != nil {
+		return err
+	}
+
+	var delayedSetupProfiles bool
+	if err := task.Get("delayed-setup-profiles", &delayedSetupProfiles); err != nil && err != state.ErrNoState {
+		return err
+	}
+	if delayedSetupProfiles {
+		logger.Debugf("Connect undo handler: skipping setupSnapSecurity for snaps %q and %q", connRef.PlugRef.Snap, connRef.SlotRef.Snap)
```

Comment:
Did you mean to `return` here?

Reply:
Thanks for spotting! Indeed, and I got the associated test wrong assuming the 2 security backend calls were from connect, which of course makes no sense, we should get 0 backend calls for connect and its undo with this flag. Fixed.